jekyll/_posts/guides/2018-09-21-usage-of-the-matrix-js-sdk.md

```diff
@@ -0,0 +1,198 @@
+});
+```
+
+In any case, once logged in either with a password or an access token, the client can get the current access token via:
+
+```javascript
+console.log(client.getAccessToken());
+```
+
+## Sync and Listen
+
+Let's perform a first sync, and listen for the response:
+
+```javascript
+client.once('sync', function(state, prevState, res) {
+    //console.log(res);
+    if(state === 'PREPARED') {
+        console.log("prepared");
+    } else {
+        console.log(state);
+        process.exit(1);
+    }
+});
+```
+
+Once the sync is complete, we can add listeners for events. To listen to ALL incoming events:
+
+```javascript
+client.on("event", function(event){
+    console.log(event.getType());
+    console.log(event);
+})
+```
+
+But that would be a lot of traffic, and too much for our simple example. Let's just listen to events happening on the timeline of rooms for which our user is a member:
+
+```javascript
+client.on("Room.timeline", function(event, room, toStartOfTimeline) {
+    console.log(event.event);
+});
+```
+
+## Access the Store
+
+When we created a new client with `sdk.createClient()`, an instance of the default store, `MatrixInMemoryStore` was created and enabled.
+
+To access it, we use `client.store`. For example, to get a list of rooms in which our user is joined:
+
+```javascript
+// client.store.rooms is an object with room IDs as property keys
+var roomIds = Object.keys(client.store.rooms);
+roomIds.forEach(roomId => {
+    console.log(roomId);
+});
+```
+
+More usefully, we could get a list of members for each of these rooms:
+
+```javascript
+var roomIds = Object.keys(client.store.rooms);
+roomIds.forEach(roomId => {
+    var members = client.store.rooms[roomId].getJoinedMembers();
+    members.forEach(member => {
+        console.log(member.name);
+    });
+});
+```
+
+For each room, we can inspect the timeline in the store:
+
+```javascript
+var roomIds = Object.keys(client.store.rooms);
+roomIds.forEach(roomId => {
+    client.store.rooms[roomId].timeline.forEach(t => {
+        console.log(JSON.stringify(t.event.content));
+    });
+});
+```
+
+## Send messages to rooms
+
+To send a message, we create a content object, and specify a room to send to:
+
+```javascript
+var content = {
+    "body": "Hello World",
+    "msgtype": "m.text"
+};
+
+client.sendEvent("!jhpZBTbckszblMYjMK:matrix.org", "m.room.message", content, "", (err, res) => {
+    console.log(err);
+});
+```
+
+Knowing this, we can build a bot which just echos back any message starting with a "!"
+
+```javascript
+var testRoomId = "!jhpZBTbckszblMYjMK:matrix.org";
+
+client.on("Room.timeline", function(event, room, toStartOfTimeline) {
+    // we know we only want to respond to messages
+    if (event.getType() !== "m.room.message") {
+        return;
+    }
+
+    // we are only intested in messages from the test room, which start with "!"
+    if (event.event.room_id === testRoomId && event.event.content.body[0] === '!') {
```

Comment:
You should avoid accessing the `event` object directly - `event.getRoomId()` is what you want here.

Reply:
...and `event.getContent().body[0]`